### PR TITLE
Fiodor Borisiuk - 0

### PR DESCRIPTION
**Pull request by fedorlol**
## Name
Fiodor Borisiuk

## Homework#
0

## Link to video with demo
https://streamable.com/sohqs

## Comment
Levels 1, 2 and 3.

### CODE REVIEW

File: Fiodor Borisiuk/0/run.rb [hunk 1]
```diff
@@ -0,0 +1,132 @@
+require 'roo'
+
+file_list = Dir.entries('.')
+xls_list = []
+file_list.each do |file|
```
Comment (taleh007) [inline Fiodor Borisiuk/0/run.rb]: Read docs about Enumerable. Use `each_with_object`

File: Fiodor Borisiuk/0/run.rb [hunk 2]
```diff
@@ -0,0 +1,132 @@
+require 'roo'
+
+file_list = Dir.entries('.')
```
Comment (taleh007) [inline Fiodor Borisiuk/0/run.rb]: How did you download `.xslx` files? Where is script?

Comment (fedorlol) [reply Fiodor Borisiuk/0/run.rb]: added script for downloading .xlsx and .xls files to main

File: Fiodor Borisiuk/0/run.rb (hunk 1 above)
Comment (fedorlol) [reply Fiodor Borisiuk/0/run.rb]: done

File: Fiodor Borisiuk/0/downloader.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+doc = Nokogiri::HTML(html).to_s
+ind_low = doc.index('СРЕДНИЕ ЦЕНЫ НА ТОВАРЫ, РЕАЛИЗУЕМЫЕ В РОЗНИЧНОЙ СЕТИ')
+ind_high_def = doc.index('Средние цены (тарифы) на отдельные виды платных')
+doc = doc[ind_low..ind_high_def]
+
+months = %w[янв фев мар апр май июн июл авг сен окт ноя дек]
+url_table = {}
+f = true
+while f
+  ind_low = doc.index('<a href="')
```
Comment (taleh007) [inline Fiodor Borisiuk/0/downloader.rb]: css selectors will better solve you task

File: Fiodor Borisiuk/0/downloader.rb [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+  ind_high = doc.index('</a>')
+  if ind_low && ind_high
+    str = doc[ind_low..ind_high]
+    ind = str.index('за')
+    break if ind.nil?
+
+    month = months.index(str[ind + 3..ind + 5]) + 1
+
+    ind = str.index('200')
```
Comment (taleh007) [inline Fiodor Borisiuk/0/downloader.rb]: u can use Regex 

```ruby
/\d{4}/
```

File: Fiodor Borisiuk/0/run.rb [hunk 5]
```diff
@@ -0,0 +1,146 @@
+require 'bundler/setup'
+Bundler.require(:default)
```
Comment (taleh007) [inline Fiodor Borisiuk/0/run.rb]: I like this line :)

File: Fiodor Borisiuk/0/run.rb [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+  arr.append('./.data/' + file) if file.index('xls')
+end
+
+xls_table = []
+xls_list.each do |xls|
+  ind = xls.index('20')
+  xls_table.append([xls, xls[ind - 3..ind + 3]])
+end
+
+@table = {}
```
Comment (taleh007) [inline Fiodor Borisiuk/0/run.rb]: useless `@` without class

File: Fiodor Borisiuk/0/run.rb [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+
+  price_arr = []
+  @frow.upto(sheet.last_row) do |i|
+    cur_price = []
+    1.upto(sheet.last_column) do |j|
+      cur_cell = sheet.cell(i, j).to_s
+      next if cur_cell.nil?
+
+      cur_cell = (cur_cell.to_f / 10_000).to_s\
+        if cur_cell.to_f != 0.0 && date[5..6].to_f <= 16
```
Comment (taleh007) [inline Fiodor Borisiuk/0/run.rb]: very strange code

File: Fiodor Borisiuk/0/run.rb [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+    end
+    if cur_price[-1].to_s == cur_price[-1].to_f.to_s
+      cur_price[0] = cur_price[0].downcase.capitalize
+      price_arr.append(cur_price)
+    end
+  end
+  @table[date] = price_arr
+end
+
+def find_by_kwd(arr, kwd)
```
Comment (taleh007) [inline Fiodor Borisiuk/0/run.rb]: `kwd` ? what is it

File: Fiodor Borisiuk/0/run.rb [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+  ans = []
+  arr.each do |i|
```
Comment (taleh007) [inline Fiodor Borisiuk/0/run.rb]: each_with_object

File: Fiodor Borisiuk/0/run.rb [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+  arr.each do |line|
+    next if line[0] == name
+
+    ans.append("'" + line[0] + "'")\
+      if ((price - eps)..(price + eps)).cover?(line[-3].to_f)
+  end
+  ans
+end
+
+def m_atoi(arr)
```
Comment (taleh007) [inline Fiodor Borisiuk/0/run.rb]: what does it mean? use understandable names

File: Fiodor Borisiuk/0/Gemfile [hunk 11]
```diff
@@ -0,0 +1,6 @@
+source 'https://rubygems.org'
+source 'http://gems.github.com'
+
```
Comment (taleh007) [inline Fiodor Borisiuk/0/Gemfile]: setup ruby

File: Fiodor Borisiuk/0/run.rb [hunk 12]
```diff
@@ -0,0 +1,146 @@
+require 'bundler/setup'
+Bundler.require(:default)
+
+file_list = Dir.entries('./.data')
+xls_list = []
+file_list.each_with_object(xls_list) do |file, arr|
```
Comment (taleh007) [inline Fiodor Borisiuk/0/run.rb]:
```suggestion
xls_list = file_list.each_with_object([]) do |file, arr|
```

**Pull request merged**